python/ray/tune/trial_runner.py

```diff
@@ -508,6 +508,9 @@ def _process_trial(self, trial):
 
             self._total_time += result[TIME_THIS_ITER_S]
 
+            if not is_duplicate:
+                trial.result_logger.on_result(result)
```

Comment:
Actually, I'm not totally sure that this works; this requires the `done` field to be set, which only occurs after a decision is made by the TrialScheduler.

Reply:
Looked through the changes; I think this is still true